src/ipa_pc/data_structures.rs

```diff
@@ -51,83 +107,89 @@ impl<G: AffineCurve> PCUniversalParams for UniversalParams<G> {
 /// polynomial.
 #[derive(Derivative)]
 #[derivative(
-    Default(bound = ""),
     Hash(bound = ""),
     Clone(bound = ""),
     Debug(bound = ""),
     Eq(bound = ""),
     PartialEq(bound = "")
 )]
 #[derive(CanonicalSerialize, CanonicalDeserialize)]
-pub struct CommitterKey<G: AffineCurve> {
-    /// The key used to commit to polynomials.
-    pub comm_key: Vec<G>,
+pub struct CommitterKey<G: Curve> {
+    /// The base point vector used for the coefficients of the polynomial.
+    pub comm_key: Vec<G::AffineRep>,
 
-    /// A random group generator.
+    /// The base point used by the opening proof to reduce to a non-blinded
+    /// opening assertion.
     pub h: G,
 
-    /// A random group generator that is to be used to make
-    /// a commitment hiding.
+    /// The base point that is to be used for hiding.
     pub s: G,
 
     /// The maximum degree supported by the parameters
     /// this key was derived from.
     pub max_degree: usize,
```

Comment:
There actually is no need to keep track of the max_degree of the parameter set it was derived from. Let us use this is the maximum degree of the current key.

Reply:
Removed max_degree parameter at all. The function degree() will simply return self.comm_key.len() - 1